src/scheduler/parse.c

```diff
@@ -1161,3 +1181,57 @@ preempt_cmp(const void *p1, const void *p2)
 	}
 }
 
+/**
+ * @brief
+ *	       Frees the memory for the corresponding sort_info structure variable given.
+ *
+ * @param[in]	si_type	- type of sort_info structure
+ *
+ * @return void
+ *
+ */
+void
+free_sort_info(enum sort_info_type si_type)
+{
+	struct sort_info *so_info;
+
+	switch (si_type) {
+		case PRIME_SORT:
+			so_info = conf.prime_sort;
+			break;
+		case NON_PRIME_SORT:
+			so_info = conf.non_prime_sort;
+			break;
+		case PRIME_NODE_SORT:
+			so_info = conf.prime_node_sort;
+			break;
+		case NON_PRIME_NODE_SORT:
+			so_info = conf.non_prime_node_sort;
+			break;
+		default:
+			return;
+	}
+
+	if (so_info != NULL) {
+		int i;
+		for (i = 0; i < MAX_SORTS; i++) {
+			if (so_info[i].res_name != NULL) {
+				free(so_info[i].res_name);
+				if (si_type == PRIME_SORT || si_type == PRIME_NODE_SORT) {
+					if (si_type == PRIME_SORT) {
+						if (so_info[i].res_name != conf.non_prime_sort[i].res_name)
```

Comment:
Yes. But here we are just assigning it to NULL so that if same memory location is referred from an another sort_info structure we should not accidentally free it or in other words to avoid double free.